### PR TITLE
feat(gba): improve suite coverage and pass arm/thumb/nes gates

### PR DESCRIPTION
**Pull request by rmstdope**
## Summary
This PR extends and stabilizes GBA automated suite coverage.

- hardens the GBA suite runner bootstrap and diagnostics
- completes remaining ARM block transfer semantics for suite compatibility
- fixes THUMB misaligned halfword and format 15 edge-case behavior
- models ARM7TDMI three-stage prefetch to match self-modifying code pipeline behavior
- adds the gba-tests memory suite to integration coverage

## Files changed
- Cargo.toml
- src/gba/console/gba.rs
- src/gba/cpu/arm.rs
- src/gba/cpu/arm7tdmi.rs
- src/gba/cpu/registers.rs
- src/gba/cpu/thumb.rs
- src/gba/integration_tests/gba_suite_runner.rs
- src/gba/integration_tests/gba_suite_tests.rs

## Validation
- cargo test gba_suite_arm_rom_passes -- --nocapture
- cargo test gba_suite_thumb_rom_passes -- --nocapture
- cargo test gba_suite_nes_rom_passes -- --nocapture
- cargo test arm_prefetch_keeps_two_instructions_after_self_modify -- --nocapture

All of the above pass locally.

### CODE REVIEW

File: src/gba/integration_tests/gba_suite_runner.rs [hunk 1]
```diff
@@ -62,59 +71,144 @@ pub fn run_suite(suite: Suite) -> SuiteResult {
...
-        if stable_pc_count >= IDLE_PC_STABLE_THRESHOLD {
-            return result_from_register(&gba, suite, cycles, pc, ExitReason::IdleLoopDetected);
+        let opcode = gba.bus_mut().peek32(pc);
+        if is_arm_branch_to_self(opcode, pc) {
+            let reason = if is_bios_exception_vector(pc) {
+                ExitReason::ExceptionVectorTrap
+            } else {
+                ExitReason::IdleLoopDetected
+            };
+            return result_from_register(&mut gba, suite, cycles, pc, reason);
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/gba/integration_tests/gba_suite_runner.rs]: Applied in 18954c6a.

I updated the suite runner to avoid peeking opcodes on every tick. It now tracks repeated PC values and only performs peek32 once the PC is stable for a short threshold, then checks for the self-branch idle signature.

This preserves behavior while reducing bus work in the hot loop.
